### ISSUE TITLE
Move high level reconstruction out of CAFMaker fcl

### ISSUE THREAD

**Issue by etyley** [SBNSoftware/sbncode#164]
The CAFMaker fcl files currently run a large number of high-level reconstruction that is probably better suited to be run in the reco2 stage: https://github.com/SBNSoftware/sbncode/blob/5d1874962dba51a7ebab9eb6c8e7ef63498f81e1/sbncode/CAFMaker/cafmakerjob_sbnd.fcl#L115-L119

As these producers are stable, this would massively expedite the procedure for re-making CAFs from the reco2 files.

**Comment by PetrilloAtWork** [comment]
I think this deserves a wider discussion toward an agreement about what each "stage" contains, how to decide where additions belong and how to call them.
Characters in the play include the people managing the workflows (for ICARUS, @SFBayLaser; for SBND... @absolution1?), people writing and exercising the configurations (plenty: @gputnam, @etyley, @pgreen135, @cjbackhouse, and I am sure I am missing things), and production groups (Mateus Carneiro, @mayawos, François Drielsma, ...), analysis infrastructure (@wesketchum, @jzennamo). Etc.

**Comment by cjbackhouse** [comment]
I am all in favour of this. My ideal is that the cafmaker job only runs the cafmaker module itself, and all the other products it needs are already in the art files. This is probably impractical for a variety of reasons, but it indicates a direction of travel. At the moment, making CAFs is pretty onerous due to all the other stuff that has to happen too.

**Comment by etyley** [comment]
I agree with Chris that is what we should aspire to. Practically speaking, I see no reason that all of the producers mentioned above (apart from cafmaker itself obviously) could not be moved to the reco2 stage. I guess there is also a question about whether we would want the systematic weights to also be run prior to the CAFMaker (e.g. run the GENIE/flux weights at the gen stage).

**Comment by wesketchum** [comment]
@PetrilloAtWork @FernandaPsihas this is something we should make happen by 2021C production? (? == .)

**Comment by PetrilloAtWork** [comment]
I'll need to pull in the ICARUS workflow manager (@SFBayLaser).

**Comment by wesketchum** [comment]
@PetrilloAtWork @SFBayLaser I can check the fcl files too, but I believe this has been done, yes?

**Comment by wesketchum** [comment]
Believe this is done. Closing.
